deps/chakrashim/core/lib/Jsrt/JsrtDebuggerObject.cpp

```diff
@@ -506,7 +506,13 @@ bool JsrtDebuggerStackFrame::Evaluate(Js::ScriptContext* scriptContext, const ch
             resolvedObject.typeId = Js::JavascriptOperators::GetTypeId(resolvedObject.obj);
             JsrtDebuggerObjectBase::CreateDebuggerObject<JsrtDebuggerObjectProperty>(this->debuggerObjectsManager, resolvedObject, scriptContext, [&](Js::Var marshaledObj)
             {
-                *evalResult = (Js::DynamicObject*)marshaledObj;
+                Js::DynamicObject *debuggerObj = (Js::DynamicObject*)marshaledObj;
+                if (returnByValue)
+                {
+                    JsrtDebugUtils::AddPropertyToObject(debuggerObj, JsrtDebugPropertyId::value, resolvedObject.obj, scriptContext);
```

Comment:
I've been looking into this a bit more.  I think the change will be a bit more invasive since CreateDebuggerObject is a template for four different types of debugger object.  That means that the parameter would need to percolate down into each class.  I'll give it a try this week, but this may be the cleaner approach for now.